samples/snippets/src/main/java/com/example/spanner/ChangeStreamSample.java

```diff
@@ -0,0 +1,626 @@
+  static void insertRows(DatabaseClient client) {
+    client.write(
+        ImmutableList.of(
+        Mutation.newInsertOrUpdateBuilder("Singers")
+          .set("SingerId").to(1)
+          .set("FirstName").to("Melissa")
+          .set("LastName").to("Garcia")
+          .build(),
+        Mutation.newInsertOrUpdateBuilder("Singers")
+          .set("SingerId").to(2)
+          .set("FirstName").to("Dylan")
+          .set("LastName").to("Shaw")
+          .build()
+      )
+    );
+  }
+
+  // Execute a change stream query, return and print out the result records.
+  static List<ChangeStreamRecord> executeChangeStreamQueryAndPrint(
+      DatabaseClient dbClient, String startTimestamp, String endTimestamp, String partitionToken) {
+    System.out.println("Executing a change stream query with: "
+        + "start_timestamp => " + startTimestamp
+        + ", end_timestamp => " + endTimestamp
+        + ", partition_token => " + partitionToken
+        + ", heartbeat_milliseconds => 5000"
+        + ", read_options => null");
+
+    final String query =
+        "SELECT * FROM READ_ChangeStreamAll ("
+        + "start_timestamp => @startTimestamp,"
+        + "end_timestamp => @endTimestamp,"
+        + "partition_token => @partitionToken,"
+        + "read_options => null,"
+        + "heartbeat_milliseconds => @heartbeatMillis"
+        + ")";
+
+    final ResultSet resultSet =
+        dbClient
+        .singleUse()
+        .executeQuery(
+          Statement.newBuilder(query)
+            .bind("startTimestamp").to(Timestamp.parseTimestamp(startTimestamp))
+            .bind("endTimestamp").to(Timestamp.parseTimestamp(endTimestamp))
+            .bind("partitionToken").to(partitionToken)
+            .bind("heartbeatMillis").to(5000)
+            .build());
+
+    ChangeStreamRecordMapper changeStreamRecordMapper = new ChangeStreamRecordMapper();
+    List<ChangeStreamRecord> res = new LinkedList<>();
+    while (resultSet.next()) {
+      // Parses result set into change stream result format.
+      final List<ChangeStreamRecord> records =
+          changeStreamRecordMapper.toChangeStreamRecords(resultSet.getCurrentRowAsStruct());
+
+      // Prints out all the query results.
+      for (final ChangeStreamRecord record : records) {
+        if (record instanceof DataChangeRecord) {
+          System.out.println("Received a DataChangeRecord: " + record);
+        } else if (record instanceof HeartbeatRecord) {
+          System.out.println("Received a HeartbeatRecord: " + record);
+        } else if (record instanceof ChildPartitionsRecord) {
+          System.out.println("Received a ChildPartitionsRecord: " + record);
+        } else {
+          throw new IllegalArgumentException("Unknown record type " + record.getClass());
+        }
+      }
+      res.addAll(records);
+    }
+
+    return res;
+  }
+
+  // Models of change stream result format.
+  interface ChangeStreamRecord extends Serializable {}
```

Comment:
Do we need these models? Or could we just use the raw structs? If we could use the result set raw structs I think we could simplify the code / remove the mapper.

Reply:
It does make the code a bit long, but if we just print out the struct, user still have to figure out how to parse the struct and extract the records in their app code, and this is non-trivial. 

On a separate note, It seems that these models are pretty common code for whoever uses raw change stream API, should we package these models somewhere so that user (and connector) can add them as dependencies if they want to use raw API?

Reply:
> It does make the code a bit long, but if we just print out the struct, user still have to figure out how to parse the struct and extract the records in their app code, and this is non-trivial.

Agreed, but the issue I see with this is that the samples need to be relatively simple so that we can add them to our documentation (like [here](https://cloud.google.com/spanner/docs/getting-started/java)). If we have these many lines of code it might be hard for us to add those there. I was thinking that we could provide a sample which only extracts the parts of the records from a struct that we care about reading for the sample. We can then refer the user to the Struct full specification which should be published somewhere else. WDYT?

> On a separate note, It seems that these models are pretty common code for whoever uses raw change stream API, should we package these models somewhere so that user (and connector) can add them as dependencies if they want to use raw API?

They are common, but I don't think they should live in the client libraries as they can evolve separately (client libraries should be a lower level layer them the models). For instance, there might be customers who are only interested in parsing some of the fields. If we were to have these in a library, IMHO they should leave in a higher abstraction library that uses the client libraries. To be honest, I don't think we should care about this at this moment. Open to discuss it further if you have a stronger opinion otherwise.